### PR TITLE
Add reference doc for dictionary/btree/hash/cased/uncased.

### PR DESCRIPTION
**Pull request by baldersheim**
@geirst @toregge @kkraune PR

### CODE REVIEW

**Review by kkraune — APPROVED** [review]
Great!

File: en/reference/schema-reference.html [hunk 1]
```diff
@@ -1283,6 +1284,18 @@ <h2 id="alias">alias</h2>
 </p>
 
 
+<h2 id="dictionary">dictionary</h2>
+<p>Contained in <code><a href="#field">field</a></code>.
+Specifies details of the dictionary. Applies only to <a href="#attribute">attributes</a> annotated with <code>fast-search</code>.
+You can specify either <code>btree</code> or <code>hash</hash>, or both.</p>
+<p>
+Note that prefix search for strings and range search for numeric fields will fallback to full scan if using <code>hash</code>.
+It is primarily intended for use when you many unique terms that have short postinglists, where the dictionary lookup cost is significant.
```
Comment (geirst) [inline en/reference/schema-reference.html:1293]:
```suggestion
It is primarily intended for use when you have many unique terms with few occurrences (short postinglists), where the dictionary lookup cost is significant.
```

File: en/reference/schema-reference.html [hunk 2]
```diff
@@ -1283,6 +1284,18 @@ <h2 id="alias">alias</h2>
...
+Normally <code>btree</code> is you best choice. This is also the default.<p>
```
Comment (geirst) [inline en/reference/schema-reference.html:1294]:
```suggestion
Normally, <code>btree</code> is your best choice. This is also the default.<p>
```

File: en/reference/schema-reference.html [hunk 3]
```diff
@@ -1283,6 +1284,18 @@ <h2 id="alias">alias</h2>
...
+<p>
+In addition you can specify if you want a folded dictionary, <code>uncased</code>, or <code>cased</code> for string attributes.
```
Comment (geirst) [inline en/reference/schema-reference.html:1296]: I don't think the term _folded dictionary_ has been used before. Can this be explained in a simple way instead?

File: en/reference/schema-reference.html [hunk 4]
```diff
@@ -1283,6 +1284,18 @@ <h2 id="alias">alias</h2>
 </p>
 
 
+<h2 id="dictionary">dictionary</h2>
+<p>Contained in <code><a href="#field">field</a></code>.
+Specifies details of the dictionary. Applies only to <a href="#attribute">attributes</a> annotated with <code>fast-search</code>.
```
Comment (geirst) [inline en/reference/schema-reference.html:1289]:
```suggestion
Specifies details of the dictionary used in the inverted index of the field. Applies only to <a href="#attribute">attributes</a> annotated with <code>fast-search</code>.
```

**Review by geirst — APPROVED** [review]
👍

File: en/reference/schema-reference.html (hunk 3 above)
Comment (baldersheim) [reply en/reference/schema-reference.html]: Done.

**Comment by baldersheim** [comment]
@jobergum FYI

**Comment by jobergum** [comment]
This renders not well. Missing code block end

**Comment by baldersheim** [comment]
Fixed in #1278

**Pull request merged**